File: src/modules/tv.ts
// ─────────────────────────────────────────────────────────────
//  DSFlix SDK  –  src/modules/tv.ts
// ─────────────────────────────────────────────────────────────

import { HttpClient } from '../http';
import {
  PaginatedResult,
  TVShow,
  TVShowDetails,
  Season,
  Episode,
  Credits,
  TVSearchOptions,
  PaginationOptions,
  MovieDetailsOptions,
  DownloadParams,
  DownloadResponse,
} from '../types';

export class TVModule {
  constructor(private http: HttpClient) {}

  /** Get TV shows ordered by popularity */
  getPopular(options?: PaginationOptions): Promise<PaginatedResult<TVShow>> {
    return this.http.get('/api/v2/tv/popular', options as Record<string, string | number | boolean | undefined>);
  }

  /** Get full details for a TV show */
  getDetails(seriesId: number, options?: MovieDetailsOptions): Promise<TVShowDetails> {
    return this.http.get(`/api/v2/tv/${seriesId}`, options as Record<string, string | number | boolean | undefined>);
  }

  /** Get all episodes for a specific season */
  getSeason(seriesId: number, seasonNumber: number): Promise<Season> {
    return this.http.get(`/api/v2/tv/${seriesId}/season/${seasonNumber}`);
  }

  /** Get details for a specific episode */
  getEpisode(seriesId: number, seasonNumber: number, episodeNumber: number): Promise<Episode> {
    return this.http.get(`/api/v2/tv/${seriesId}/season/${seasonNumber}/episode/${episodeNumber}`);
  }

  /** Get cast and crew credits for a TV season */
  getCredits(seriesId: number, seasonNumber: number): Promise<Credits> {
    return this.http.get(`/api/v2/tv/${seriesId}/season/${seasonNumber}/credits`);
  }

  /** Get full cast for a TV show */
  getCast(seriesId: number): Promise<Credits> {
    return this.http.get(`/api/v2/tv/${seriesId}/cast`);
  }

  /** Search TV shows by name */
  search(query: string, options?: TVSearchOptions): Promise<PaginatedResult<TVShow>> {
    return this.http.get('/api/v2/search/tv', { query, ...options } as Record<string, string | number | boolean | undefined>);
  }

  /** Request a download link for a TV episode */
  download(params: DownloadParams): Promise<DownloadResponse> {
    return this.http.post('/api/v2/download', params);
  }
}

File: python/dsflix_sdk/modules/tv.py
from __future__ import annotations

from ..http import HttpClient


class TVModule:
    def __init__(self, http: HttpClient) -> None:
        self.http = http

    def get_popular(self, page: int | None = None, language: str | None = None) -> dict:
        return self.http.get("/api/v2/tv/popular", {"page": page, "language": language})

    def get_details(self, series_id: int, language: str | None = None, append_to_response: str | None = None) -> dict:
        return self.http.get(f"/api/v2/tv/{series_id}", {"language": language, "append_to_response": append_to_response})

    def get_season(self, series_id: int, season_number: int) -> dict:
        return self.http.get(f"/api/v2/tv/{series_id}/season/{season_number}")

    def get_episode(self, series_id: int, season_number: int, episode_number: int) -> dict:
        return self.http.get(f"/api/v2/tv/{series_id}/season/{season_number}/episode/{episode_number}")

    def get_credits(self, series_id: int, season_number: int) -> dict:
        return self.http.get(f"/api/v2/tv/{series_id}/season/{season_number}/credits")

    def get_cast(self, series_id: int) -> dict:
        return self.http.get(f"/api/v2/tv/{series_id}/cast")

    def search(self, query: str, page: int | None = None, language: str | None = None, include_adult: bool | None = None) -> dict:
        return self.http.get(
            "/api/v2/search/tv",
            {"query": query, "page": page, "language": language, "include_adult": include_adult},
        )

    def download(self, media_id: int, quality: str | None = None) -> dict:
        return self.http.post("/api/v2/download", {"media_id": media_id, "quality": quality})
